client/state/sites/blog-stickers/reducer.js

```diff
@@ -31,7 +35,20 @@ export const items = createReducer(
 				[ blogId ]: compact( concat( stickerName, state[ blogId ] ) ),
 			};
 		},
-	},
+		[ SITES_BLOG_STICKER_REMOVE ]: ( state, action ) => {
+			const { blogId, stickerName } = action.payload;
+
+			// If the blog doesn't have this sticker, do nothing
+			if ( ! includes( state[ blogId ], stickerName ) ) {
+				return state;
+			}
+
+			return {
+				...state,
+				[ blogId ]: compact( filter( state[ blogId ], sticker => sticker !== stickerName ) ),
```

Comment:
i don't think that we need `compact` in this case.
Also would be cool to flip `filter` into a `reject` so you make the positive assertion.